### ISSUE TITLE
Test multi-label task in 'mmcv/engine/test.py'

### ISSUE THREAD

**Issue by MaxeeCR** [open-mmlab/mmcv#1873]
**Checklist**
1. I have searched related issues but cannot get the expected help.
2. I have read the FAQ documentation but cannot get the expected help.

Hi, I have a question about training multi-label task. When the output is `List` type e.g.
```
...
def forward(self, x):
        multi_label_scores = []
        for idx in range(n):
            single_label_score = self.mutli_class_fc_layers[idx](x)
            multi_label_scores.append(single_label_score)
        return multi_label_scores
...
```
In testing phase, the output of single sample will be `extend` rather than `append` to `results` parameter. This will result to wrong size of `results` which may cause error in subsequent modules.
> File: mmcv/engine/test.py  
> Line: 73
```
    for i, data in enumerate(data_loader):
        with torch.no_grad():
            result = model(return_loss=False, **data)
      --->  results.extend(result)
```

Is there any method to solve this problem? Thx

**Comment by zhouzaida** [comment]
The data is batched data, so `forward` should return a list of `multi_label_scores`.

**Comment by MaxeeCR** [comment]
Thx, i solved by other method

> The data is batched data, so `forward` should return a list of `multi_label_scores`.
